### PR TITLE
j: add devel package

### LINKED ISSUE #32640: Package `j` provides `libj.so`, but it's not linked or copied into the default linker search path
### System

* xuname:  `Void 5.13.9_1 x86_64 GenuineIntel uptodate rrrmFFFF`
* package:  `j-901.f_1`

### Expected behavior

`libj.so` is available on the default dynamic linker search path (ie, a system call like `dlopen("j")` should locate `libj.so`)

### Actual behavior

`libj.so` is present and functional at `/usr/lib64/j/j64/libj.so`, but not available in `/usr/lib64`.

### Steps to reproduce the behavior

Attempt to use dynamic linking to access the `libj.so` file provided in the J distribution.

Here's a minimal C program that demonstrates the difference (obviously, the `janet` package is installed on my system!):

```c
#include <dlfcn.h>
#include <stdio.h>

int main(int argc, char **argv) {
  void *p = dlopen("libjanet.so", RTLD_NOW);
  printf("Janet DLL pointer: %p\n", p);
  void *q = dlopen("libj.so", RTLD_NOW);
  printf("J DLL pointer: %p\n", q);
}
```

Here's the execution example on my system:

```
Desktop [] ⊕ gcc ex.c -ldl                                                                                              
Desktop [] ⊕ ./a.out                                                                                                    
Janet DLL pointer: 0x5561857862c0
J DLL pointer: (nil)
```

### PR DESCRIPTION
**Pull request by subsetpark**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

---

fixes #32640.

### CODE REVIEW

File: srcpkgs/j-devel/template [hunk 1]
```diff
@@ -0,0 +1 @@
+../j/template
```
Comment (Duncaen) [inline srcpkgs/j-devel/template:1]: This should just link to the main packages directory, `ln -s j srcpkgs/j-devel`

Comment (subsetpark) [reply srcpkgs/j-devel/template]: Hi @Duncaen, I interpreted your comment as `ln -s ./j j-devel` - but please correct me if I've misunderstood. Thanks for your help!

Comment (Duncaen) [reply srcpkgs/j-devel/template]: We generally just use the package name, i.e. `ln -s j j-devel` if you do it form the srcpkgs directory.
This makes it easier to just read the source package without having to resolve or parse any paths.

**Comment by subsetpark** [comment]
@ericonr done!

**Comment by ericonr** [comment]
Hi, can you rebase the PR so CI can run?

**Comment by subsetpark** [comment]
@ericonr done.

**Comment by ericonr** [comment]
And now apparently the build errors out >.<

Can you try adding `CFLAGS=-Wno-error` to the template? Otherwise it might be necessary to patch `-Werror` out from their stuff.

**Comment by subsetpark** [comment]
@ericonr unfortunately, I couldn't find a way to add that to CFLAGS in the makefile. I think I've reached the end of my abilities on this one; it gets into specifics of the build script. Hopefully the maintainer of the template understands better how to get this over the finish line.

File: srcpkgs/j/template [hunk 2]
```diff
@@ -139,3 +139,14 @@ do_install() {
...
 }
+
+j-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
```
Comment (paper42) [inline srcpkgs/j/template:150]: None of these regexes will match, did you test this PR?
The only libraries in j are:
```
/usr/lib/j/j64/jconsole
/usr/lib/j/j64/libj.so
/usr/lib/j/j64avx/jconsole
/usr/lib/j/j64avx/libj.so
/usr/lib/j/j64avx2/jconsole
/usr/lib/j/j64avx2/libj.so
```
and they are required in the main package or jc won't work. What exactly do you want to split to -devel?

**Comment by paper42** [comment]
As mentioned in https://github.com/void-linux/void-packages/pull/32644#discussion_r701343215, this approach is wrong. The files that are supposed to be split to a -devel subpackage do not exist, so there is no need to do the split.

**Pull request closed without merging**